compass/ocean/mesh/remap_topography.py

```diff
@@ -94,64 +108,203 @@ def run(self):
         Run this step of the test case
         """
         config = self.config
+        weight_generator = config.get('remap_topography', 'weight_generator')
+
+        self._create_target_scrip_file()
+        if weight_generator == 'tempest':
+            self._partition_scrip_file('source.scrip.nc')
+            self._partition_scrip_file('target.scrip.nc')
+            self._create_weights_tempest()
+        elif weight_generator == 'esmf':
+            self._create_weights_esmf()
+        else:
+            msg = f'Unsupported weight generator function {weight_generator}'
+            raise ValueError(msg)
+        self._remap_to_target()
+        self._modify_remapped_bathymetry()
+
+    def _create_target_scrip_file(self):
+        """
+        Create target SCRIP file from MPAS mesh file.
+        """
+        logger = self.logger
+        logger.info('Create source SCRIP file')
+
+        config = self.config
+        section = config['remap_topography']
+        expandDist = section.getfloat('expandDist')
+        expandFactor = section.getfloat('expandFactor')
+
+        descriptor = MpasCellMeshDescriptor(
+            fileName='base_mesh.nc',
+            meshName=self.mesh_name,
+        )
+        descriptor.to_scrip(
+            'target.scrip.nc',
+            expandDist=expandDist,
+            expandFactor=expandFactor,
+        )
+
+        logger.info('  Done.')
+
+    def _partition_scrip_file(self, in_filename):
+        """
+        Partition SCRIP file for parallel mbtempest use
+        """
         logger = self.logger
-        parallel_executable = config.get('parallel', 'parallel_executable')
+        logger.info('Partition SCRIP file')
+
+        # Convert to NetCDF3 64-bit
+        args = [
+            'ncks', '-5',
+            in_filename,
+            in_filename.replace('.nc', '.64bit.nc'),
```

Comment:
I'm not particularly fond of these `relpace()` calls.  Could we instead make all the filenames at the top of this method?  Could we extract the `stem` and then use f-strings to make the filenames, e.g.:
```
        stem = pathlib.Path(in_filename).stem
        64bit_filename = f'{stem}.64bit.nc'
        h5m_filename = f'{stem}.64bit.h5m'
        part_filename = f'{stem}.64bit.p{self.ntasks}.h5m'
```
It's unclear to me if the `.64bit` suffix is meaningful for the `.h5m` files and I'd suggest removing it.